CONTRIBUTING.md

```diff
@@ -0,0 +1,140 @@
+- Use `ensureNativeComponent` instead of `requireNativeComponent` to ensure the underlying native component is properly memoized and only imported once.
+- You will need to decide what is a token and what is a property, and how you want to create your component's API surface and map it to the native view's API.
+- If you want the default values of your component's props / tokens to come from the Native Module, you can use the `constantsToExport` API, and pass it to `slotProps` in your component. You can see an example of this in `experimental-avatar`.
+
+## Creating new native apple components
+
+This section is specifically focused on creating new components for Apple platforms. Most if not all of these are wrapped controls from [FluentUI Apple](https://github.com/microsoft/fluentui-apple).
+
+If you are creating a new component from scratch (AKA your wrapped FluentUI Apple control currently has no analog in FluentUI React Native), you have the most leeway to design your API. In general, you will probably want to expose each property from the native swift control as either a token or prop (or both) to JS.
+
+If you want to add a native implementation to an existing component, you may be better off making a duplicate "experimental" component that contains the native implementation, as using a Native Module limits you to one slot, where an existing JS component may have multiple slots. Examples of duplicate native implementation include "experimental-avatar" and "experimental-native-button".
+
+An example of how we would create Shimmer with the compose framework if all the swift properties on the native control were tokens would look like so. This is how it's currently implemented
+
+```
+export type ShimmerTokens = {
+  // add the native swift properties here
+}
+// the native control on the inside needs to receive these
+export type NativeShimmerProps = ShimmerTokens & ViewProps;
+// the shimmer control only should get these via customization
+export type ShimmerProps = ViewProps
+// tokensThatAreAlsoProps is false and can be omitted because nothing is shared
+// type looks like this:
+props: ShimmerProps;
+slotProps: { root: NativeShimmerProps };
+tokens: ShimmerTokens;
+// buildProps would look like this:
+root: buildProps(tokens => ({ ...tokens }), [])
+// this is because the tokens aren't depending on any props
+// which means they only change when the theme is updated
+```
+
+If instead, some of the native swift properties were tokens AND props, it would look something like this:
+
+```typescript
+const tokensThatAreAlsoProps: (keyof ShimmerTokens)[] = [/* list all the tokens here */];
+
+// in the body of the component then add
+tokenProps,
+
+// then buildProps becomes:
+  root: buildProps(tokens => ({...tokens}), tokenProps)
+
+```
+
+To add a native module that wraps a FluentUI Apple control:
```

Comment:
@HeyImChris is your video somewhere in public that we can link for additional reference?

Reply:
I still don't know where the recorded videos are going. I just now got write permissions to the repo to upload my code to a shared repo

Reply:
Here's the macOS project though https://dev.azure.com/microsoft/ReactNative/_git/RNSummit2021?version=GBmain&path=%2FProgressBarApp_macOS

Reply:
The iOS documentation for native modules on react native core's website I feel is sufficient.  Especially because this section is both for iOS and macOS.

Reply:
My repo and video recording are more accurately used for react-native anyway since I don't mention FURN at all, which might confuse someone a little bit if they trying applying it to this dependent of RN